MapboxDirections.xcodeproj/project.pbxproj

```diff
@@ -1360,7 +1360,8 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.mapbox.MapboxDirections;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
-				SWIFT_VERSION = 3.0;
+				SWIFT_SWIFT3_OBJC_INFERENCE = Off;
+				SWIFT_VERSION = 4.0;
```

Comment:
Should we wait with this migration until the release this fall? And even then, Swift 3.2 should be sufficient. Also, I think CI bails because of this. Is the code changes compatible with Swift 3 and 4?